rotkehlchen/tests/unit/test_inquirer.py

```diff
@@ -502,3 +503,29 @@ def mock_coingecko_return(url, *args, **kwargs):  # pylint: disable=unused-argum
         inquirer.find_usd_price(A_ETH, ignore_cache=True)
         # Check that the last price query contacted coingecko api
         assert coingecko_api_calls == 2
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [False])
+@pytest.mark.parametrize('oracle', [CurrentPriceOracle.COINGECKO, CurrentPriceOracle.DEFILLAMA, CurrentPriceOracle.CRYPTOCOMPARE])  # noqa: E501
+def test_punishing_of_oracles_works(inquirer, oracle):
```

Comment:
Can you do a few things more here?

1. Add 1-2 comments around the test explaining what you do. It's easy to understand but they would help.
2. Modify the logic a bit.
    - Make ony 1 oracle fail, and the others work
    - Test that the second oracle's price is returned but that the first oracle is queried for the first 5 calls
    - At the 6th call assert that the call count of the penalized oracle did not increase and as you already did that it's penalized.
    - And then at the end keep the logic you have, that by increasing the time you end up with a non-penalized oracle.